### ISSUE TITLE
Running `.\eng\build.cmd` results in error when building repository

### ISSUE THREAD

**Issue by ShreyasJejurkar** [dotnet/aspnetcore#58484]
After installing JDK manually #58457 with Winget, I faced the next error when I tried to build the entire repo with `build.cmd` command. 
I did not pass any argument to the script because I wanted to build a debug build of the entire repo for local debugging. But I encountered the below error. 
![Image](https://github.com/user-attachments/assets/2616e811-d352-4435-9a0b-db71d7488018)

**Comment by ShreyasJejurkar** [comment]
After installing VS Community Preview 2022, the error went away. But script does not say anything what's wrong or what's missing, causing confusion.

**Comment by amcasey** [comment]
Based on the stack, it looks like it's failing to find the build tool (msbuild?) path.  Based on the follow-up comment about installing VS, my guess is that msbuild was simply not available.  We probably just need more validation in `InitializeBuildTool` (or its callers?) in `tools.ps1`.

**Comment by amcasey** [comment]
The first error I hit was actually that `Microsoft.DotNet.Arcade.MSBuild.Xcopy` 17.1.0 isn't available.  Updating the value in global.json seemed to work.  I assume it will still blow up without VS.

**Comment by amcasey** [comment]
The next error was that a particular Windows SDK wasn't installed.  I installed it, rather than installing VS.

**Comment by amcasey** [comment]
The next error was `The VCToolsInstallDir property is not defined`.  I'm not sure I can work around that one without installing VS.

**Comment by amcasey** [comment]
TL;DR: I didn't hit the originally reported issue (and so could not debug/fix it) but the correct fix is probably to detect VS earlier in the process and explicitly request it if it's not installed.

**Comment by ViktorHofer** [comment]
Hitting the original reported issue as well.

**Comment by amcasey** [comment]
@ViktorHofer With or without installing VS?

**Comment by ViktorHofer** [comment]
With VS installed. But I'm also missing a few required components. Importing the repo's .vsconfig file right now.

**Comment by ShreyasJejurkar** [comment]
> The first error I hit was actually that `Microsoft.DotNet.Arcade.MSBuild.Xcopy` 17.1.0 isn't available. Updating the value in global.json seemed to work. I assume it will still blow up without VS.

I had face that issue as well, but after I close terminal and reopened it and ran again, it got fixed somehow.

**Comment by martincostello** [comment]
FWIW I set up a brand-new computer today, and that worked for me when I tried to build everything as a test and was missing the Windows 10 SDK for the IIS bits.

**Comment by ViktorHofer** [comment]
> Did this fix your issue? 

Yes. After I installed the missing components from the imported .vsconfig file, the build succeeded. Maybe it would be possible to make that clear and throw a better error than the one that is highlighted in the initial post.

**Comment by amcasey** [comment]
> > Did this fix your issue?
> 
> Yes. After I installed the missing components from the imported .vsconfig file, the build succeeded. Maybe it would be possible to make that clear and throw a better error than the one that is highlighted in the initial post.

Yeah, I'd love to improve that error message, but I don't understand that script well enough to adjust it without debugging it.  That's why I was focused on repro'ing it.
